homeassistant/components/axis.py

```diff
@@ -0,0 +1,344 @@
+        if component in EVENT_TYPES:
+            # Sensors are created by device calling event_initialized
+            # when receiving initialize messages on metadatastream
+            device.add_event_topic(convert(component, 'type', 'subscribe'))
+            if not enable_metadatastream:
+                enable_metadatastream = True
+        else:
+            discovery.load_platform(hass, component, DOMAIN, config)
+
+    if enable_metadatastream:
+        device.initialize_new_event = event_initialized
+        device.initiate_metadatastream()
+    AXIS_DEVICES[device.serial_number] = device
+    return True
+
+
+def _read_config(hass):
+    """Read Axis config."""
+    path = hass.config.path(CONFIG_FILE)
+
+    if not os.path.isfile(path):
+        return {}
+
+    with open(path) as f_handle:
+        # Guard against empty file
+        return json.loads(f_handle.read() or '{}')
+
+
+def _write_config(hass, config):
+    """Write Axis config."""
+    data = json.dumps(config)
+    with open(hass.config.path(CONFIG_FILE), 'w', encoding='utf-8') as outfile:
+        outfile.write(data)
+
+
+def event_initialized(event):
+    """Register event initialized on metadatastream here."""
+    hass = event.device_config('hass')
+    discovery.load_platform(hass,
+                            convert(event.topic, 'topic', 'platform'),
+                            DOMAIN, {'axis_event': event})
+
+
+class AxisDeviceEvent(Entity):
+    """Representation of a Axis device event."""
+
+    def __init__(self, axis_event):
+        """Initialize the event."""
+        self.axis_event = axis_event
+        self._event_class = convert(self.axis_event.topic, 'topic', 'class')
+        self._name = '{}_{}_{}'.format(self.axis_event.device_name,
+                                       convert(self.axis_event.topic,
+                                               'topic', 'type'),
+                                       self.axis_event.id)
+        self._delay = self.axis_event.device_config(CONF_TRIGGER_TIME)
+        self._timer = None
+        self.axis_event.callback = self._update_callback
+
+    def _update_callback(self):
+        """Update the sensor's state, if needed."""
+        self.update()
+
+        if self._timer is not None:
+            self._timer()
+            self._timer = None
+
+        if self._delay > 0 and not self.is_on:
+            # Set timer to wait until updating the state
+            def _delay_update(now):
+                """Timer callback for sensor update."""
+                _LOGGER.debug("%s Called delayed (%s sec) update.",
+                              self._name, self._delay)
+                self.schedule_update_ha_state()
+                self._timer = None
+
+            self._timer = track_point_in_utc_time(
+                self.hass, _delay_update,
+                utcnow() + timedelta(seconds=self._delay))
+        else:
+            self.schedule_update_ha_state()
+
+    @property
+    def is_on(self):
```

Comment:
This is only needed for things that are based on `ToggleDevice` or `BinarySensor`. Why did you add it to your base class?